HerPortal.ManagementShell/DatabaseOperation.cs

```diff
@@ -0,0 +1,135 @@
+using HerPortal.BusinessLogic.Models;
+using HerPortal.Data;
+using Microsoft.EntityFrameworkCore;
+
+namespace HerPortal.ManagementShell;
+
+public class DatabaseOperation : IDatabaseOperation
+{
+    private readonly HerDbContext dbContext;
+    private readonly OutputProvider outputProvider;
+
+    public DatabaseOperation(HerDbContext dbContext, OutputProvider outputProvider)
+    {
+        this.dbContext = dbContext;
+        this.outputProvider = outputProvider;
+    }
+
+    public List<User> GetUsersWithLocalAuthorities()
+    {
+        return dbContext.Users
+            .Include(user => user.LocalAuthorities)
+            .ToList();
+    }
+
+    public List<LocalAuthority> GetLas(string[] custodianCodes)
+    {
+        return custodianCodes
+            .Select(code => dbContext.LocalAuthorities
+                .Single(la => la.CustodianCode == code))
+            .ToList();
+    }
+
+    public void RemoveUserOrLogError(User user)
+    {
+        using var dbContextTransaction = dbContext.Database.BeginTransaction();
+        try
+        {
+            // removing a user also deletes all associated rows in the LocalAuthorityUser table
+            dbContext.Users.Remove(user);
+            dbContext.SaveChanges();
+            var deletionConfirmation = outputProvider.Confirm(
```

Comment:
Confirmation probably wants to happen outside (and then we don't even need to roll anything back).